### ISSUE TITLE
Improve and remake all events filter handling

### ISSUE THREAD

**Issue by ayush-goyal** [HackGT/live-site#211]
For the `AllEvents` code [here](https://github.com/HackGT/live-site/blob/master/src/components/tabs/home/AllEvents.tsx), we want to do this better with Chakra UI. Currently, a lot of the code is really verbose and really overwritten, so we can simplify it a lot with Chakra UI using [form control](https://chakra-ui.com/docs/form/form-control) and other input elements. You can use react's `useState` function to keep track of the different filtering options, and update the events based on that.

Basically, we only want to show the event cards based on the filters. So for this issue, let's update this component to use Chakra UI, and make the cards and all use Chakra UI components like Box.
